### ISSUE TITLE
Upgrade CircleCI builds to at least Maven 3.6.3 

### ISSUE THREAD

**Issue by elharo** [prestodb/presto#22126]
I coaxed the CircleCI builds into telling me what version of Maven they were using to test and as I suspected from indirect evidence it was pre-3.6.1:

Error reading historical timing data: file does not exist
Requested weighting by historical based timing, but they are not present. Falling back to weighting by name.
�[1mApache Maven 3.5.4 (Red Hat 3.5.4-5)�[m
Maven home: /usr/share/maven
Java version: 1.8.0_362, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.362.b08-3.el8.x86_64/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "5.15.0-1053-aws", arch: "amd64", family: "unix"
[INFO] Scanning for projects...
[INFO]

The Github actions builds are using, I think, 3.6.3 or maybe something later. This needs to be upgraded.

**Comment by elharo** [comment]
Looks like the issue is in the docker image public.ecr.aws/oss-presto/prestocpp:root-20240129, which has Maven 3.5.4:

[root@46666eb6db42 /]# mvn -v
Apache Maven 3.5.4 (Red Hat 3.5.4-5)
Maven home: /usr/share/maven
Java version: 1.8.0_362, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.362.b08-3.el8.x86_64/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "6.6.12-linuxkit", arch: "amd64", family: "unix"

**Comment by elharo** [comment]
And that image gets it from this command:

dnf install -y java-11-openjdk 

This installs Maven 3.5.4

```
# mvn -v
Apache Maven 3.5.4 (Red Hat 3.5.4-5)
Maven home: /usr/share/maven
Java version: 1.8.0_362, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.362.b08-3.el8.aarch64/jre
```

**Comment by elharo** [comment]
fixed by #22371
